### DISCUSSION TITLE
Static build does not render stencil components

### DISCUSSION THREAD

**Discussion by SamCour** [storybookjs/storybook#27033] [Help] [answered]
### Summary

Hello everyone,

I am a bit stuck right now.

I've a storybook project that uses the webcomponent-vite base and i'm using an external component library built in Stencil, installed via npm.

When i run the dev server, everything works fine, the stories render with the components but when i build the static output and launch the index.html file, the stencil components do not render anymore.

When i navigate to the story, i get an error in the browser saying:
`TypeError: Failed to fetch dynamically imported module: http://127.0.0.1:5500/<removed>/storybook-static/assets/<removed>.entry.js undefined`

And i did verify, the <story>.entry.js files are not being generated by the compiler, for all the stories, yet the files are each time called upon by the runtime.

As it's a company project, i can't share actual code.

Appreciate any help.

Thanks.

**Comment by jarrodthibodeau** [comment, marked as answer]
@SamCour you ever find a resolution to this? 

We are now running into this same problem... If you don't care about your bundle size, the fix you can implement can be found in the StencilJS documentation here. https://stenciljs.com/docs/config-extras#enableimportinjection

That will resolve the issues but adds all your components as separate entry files. We used to have it working because we had this flag on due to using Vite but were told to turn it off to reduce the bundle size so we are now dealing with this as a side effect of that decision.

**Reply by SamCour** [reply]
Thanks mate!

What i did back then was just switch back to webpack instead of vite and that did the trick.

As its storybook, most people will not care about bundle size anyways in my opinion. Now they have a answer to the problem.

Have a great one.

**Reply by jarrodthibodeau** [reply]
@SamCour good to know! Yeah for storybook it doesn't matter but it does when you are using the library in production 😅 which is the main reason we had to turn it off.

Think I will be going the webpack route myself then.

Cheers!
